tests/plugins/test_specifications_impl.py

```diff
@@ -0,0 +1,111 @@
+from unittest import TestCase
+from tempfile import NamedTemporaryFile
+
+from aac.plugins.specifications.specifications_impl import spec_validate, AacSpecValidationException
+
+
+class TestSpecifications(TestCase):
+
+    def test_spec_validate(self):
+
+        with NamedTemporaryFile("w") as temp_spec:
+            temp_spec.write(VALID_SPEC)
+            temp_spec.seek(0)
+
+            try:
+                spec_validate(temp_spec.name)
+            except AacSpecValidationException as e:
+                self.fail("validate() raised AacSpecValidationException with error: {0}".format(e))
+
+    def test_spec_validate_fails_with_missing_requirement(self):
+
+        with NamedTemporaryFile("w") as temp_spec:
+            temp_spec.write(INVALID_SPEC_MISSING_REQ_ID)
+            temp_spec.seek(0)
+
+            with self.assertRaises(AacSpecValidationException) as context_manager:
+                spec_validate(temp_spec.name)
+
+            validation_exception = context_manager.exception
+            self.assertIn("Invalid requirement id \"SUB-3\"", validation_exception.message)
```

Comment:
I don't think `context_manager` wouldn't be active in this scope, anymore, would it? Should we have this, instead?

```suggestion
                validation_exception = context_manager.exception
                self.assertIn("Invalid requirement id \"SUB-3\"", validation_exception.message)
```

Reply:
It is accessible outside the scope. Moving this in would hide the exception from the assert if it doesn't receive and exception.

Reply:
Well, I just learned something about context managers, then. Thanks!